third-party/rsutils/src/network-adapter-watcher.cpp

```diff
@@ -0,0 +1,234 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <rsutils/os/network-adapter-watcher.h>
+#include <rsutils/shared-ptr-singleton.h>
+#include <rsutils/signal.h>
+#include <rsutils/easylogging/easyloggingpp.h>
+
+#if ! defined( __APPLE__ ) && ! defined( __ANDROID__ )
+#ifdef _WIN32
+
+#include <windows.h>
+#include <iphlpapi.h>
+#pragma comment( lib, "iphlpapi.lib" )
+
+#else
+
+#include <sys/socket.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <sys/eventfd.h>
+#include <unistd.h>
+#include <poll.h>
+
+#endif
+#endif  // ! __APPLE__ && ! __ANDROID__
```

Comment:
You can `#endif` at end of file instead of repeatingly checking this

Reply:
I still need to supply a ctor for `network_adapter_watcher`, but I'll check

Reply:
Keeping as-is